projects/ngrx.io/content/guide/router-store/selectors.md

```diff
@@ -48,3 +48,53 @@ export const selectCarsByColor = createSelector(
    (cars, params) => cars.filter(c => c.color === params['color'])
 );
 ```
+
+## Extracting all params in the current route
+
+The `selectRouteParam{s}` selector extracts params from the **leaf child** route segment only.
+
+It means that when using nested routes with the Angular router (use of the `children` property), only params from the leaf of the matching URL Tree will be extracted.
+
+If the routes are defined as:
+
+```typescript
+[
+  {
+    path: 'my/:urlPath',
+    component: /* ... */,
+    children: [
+      {
+        path: 'is/:matched',
+        component: /* ... */,
+      },
+    ],
+  },
+]
+```
+
+Using `selectRouteParam{s}` will get the `matched` param but not the `urlPath` param, because it is not located in a leaf of the URL Tree.
+
+If all params in the URL Tree need to be extracted (both `urlPath` and `matched`), the following custom selector can be used. It accumulates params of all the segments in the matched route:
+
+```typescript
+import { Params } from '@angular/router';
+import { createSelector } from '@ngrx/store';
+
+export const selectRouteNestedParams = createSelector(selectRouter, (router) => {
+  let currentRoute = router?.state?.root;
+  let params: Params = {};
+  while (currentRoute?.firstChild) {
+    currentRoute = currentRoute.firstChild;
+    params = {
+      ...params,
+      ...currentRoute.params,
+    };
+  }
+  return params;
+});
+
+export const selectRouteNestedParam = (param: string) =>
+  createSelector(selectRouteNestedParams, (params) => params && params[param]);
+```
+
+**Note:** Beware using this accumulation technique when two params with the same name exist in the route (e.g. `my/:route/:id/with/another/:id`). Only the rightmost value is accessible because leftmost values are overwritten by the rightmost one in the traversal.
```

Comment:
```suggestion
<div class="alert is-important">

Beware of using this accumulation technique when two params with the same name exist in the route (e.g. `my/:route/:id/with/another/:id`). Only the rightmost value is accessible because leftmost values are overwritten by the rightmost one in the traversal.

</div>
```